src/main/vcpf4/usbd_cdc_vcp.c

```diff
@@ -124,6 +131,12 @@ static uint16_t VCP_Ctrl(uint32_t Cmd, uint8_t* Buf, uint32_t Len)
 
       //Note - hw flow control on UART 1-3 and 6 only
       case SET_LINE_CODING:
+          // If a callback is provided, tell the upper driver of changes in baud rate
+     	 if (Buf && (Len == sizeof (*plc))) {
```

Comment:
(Ok, ST isn't that pedantic about alignment and endianness)
`Len >= sizeof (*plc)` may be safer test (from assert above)...

Reply:
If Len is greater than sizeof (*plc) then we're probably not looking at a LINE_CODING structure!

Reply:
I don't know (or want to study) USB standard in this regard. Someone written assert above that way, checking that enough data are available. Host USB implementation that sends too big wLength will cause pretty nasty bug ...